src/open_inwoner/openzaak/import_export.py

```diff
@@ -0,0 +1,146 @@
+        )
+        status_types = ZaakTypeStatusTypeConfig.objects.filter(
+            zaaktype_config__in=zaak_types
+        )
+        resultaat_types = ZaakTypeResultaatTypeConfig.objects.filter(
+            zaaktype_config__in=zaak_types
+        )
+
+        return cls(
+            catalogus_configs=catalogus_configs,
+            zaak_types=zaak_types,
+            informatie_object_types=informatie_object_types,
+            status_types=status_types,
+            resultaat_types=resultaat_types,
+        )
+
+    def as_dicts_iter(self) -> Generator[dict, Any, None]:
+        for qs in self:
+            serialized_data = serializers.serialize(
+                queryset=qs,
+                format="json",
+                use_natural_foreign_keys=True,
+                use_natural_primary_keys=True,
+            )
+            json_data: list[dict] = json.loads(
+                serialized_data,
+            )
+            yield from json_data
+
+    def as_jsonl_iter(self) -> Generator[str, Any, None]:
+        for row in self.as_dicts():
+            yield json.dumps(row)
+            yield "\n"
+
+    def as_dicts(self) -> list[dict]:
+        return list(self.as_dicts_iter())
+
+    def as_jsonl(self) -> str:
+        return "".join(self.as_jsonl_iter())
+
+
+@dataclasses.dataclass
+class CatalogusConfigImport:
+    """Import CatalogusConfig(s) and all associated relations."""
+
+    total_rows_processed: int = 0
+    catalogus_configs_imported: int = 0
+    zaaktypes_imported: int = 0
+    inormatie_object_types_imported: int = 0
+    status_types_imported: int = 0
+    resultaat_types_imported: int = 0
+
+    @classmethod
+    def from_jsonl_stream_or_string(cls, stream_or_string: io.IOBase | str) -> Self:
+        """Import objects from a stream or string contain JSONL."""
+        model_to_counter_mapping = {
+            "CatalogusConfig": "catalogus_configs_imported",
+            "ZaakTypeConfig": "zaaktypes_imported",
+            "ZaakTypeInformatieObjectTypeConfig": "inormatie_object_types_imported",
+            "ZaakTypeStatusTypeConfig": "status_types_imported",
+            "ZaakTypeResultaatTypeConfig": "resultaat_types_imported",
+        }
+
+        object_type_counts = defaultdict(int)
+        for deserialized_object in serializers.deserialize("jsonl", stream_or_string):
+            deserialized_object.save()
+            object_type = deserialized_object.object.__class__.__name__
+            object_type_counts[object_type] += 1
+
+        creation_kwargs = {
+            "total_rows_processed": sum(object_type_counts.values()),
+        }
+
+        for model_name, counter_field in model_to_counter_mapping.items():
+            creation_kwargs[counter_field] = object_type_counts[model_name]
+
+        return cls(**creation_kwargs)
+
+    @classmethod
+    def from_django_storage_jsonl(cls, file_name: str, storage: Storage):
+        """Import objects from a named JSONL file on a Django storage."""
```

Comment:
- Wouldn't it make sense to call the method "from_jsonl_in_django_storage" given this docstring?
- If the method was named "import_from_jsonl_in_django_storage", the docstring wouldn't be needed at all
Similar for `from_jsonl_stream_or_string` above.